runtime/port/common/j9sysinfo_helpers.c

```diff
@@ -210,3 +219,46 @@ getX86CPUID(uint32_t leaf, uint32_t *cpuInfo)
 #endif
 #endif
 }
+
+/**
+ * Similar to getX86CPUID() above, but with a second subleaf parameter for the
+ * leaves returned by the 'cpuid' instruction which are further divided into
+ * subleaves.
+ * 
+ * @param[in]	leaf		Value in EAX when cpuid is called to determine what info
+ * 							is returned.
+ * 				subleaf		Value in ECX when cpuid is called which is needed by some
+ * 							leafs returned in order to further specify what is returned
+ * @param[out]	cpuInfo		Reference to the an integer array which holds the data
+ * 							of EAX,EBX,ECX and EDX registers returned by cpuid.
+ * 							cpuInfo[0] holds EAX and cpuInfo[4] holds EDX.
+ */
+void
+getX86CPUIDext(uint32_t leaf, uint32_t subleaf, uint32_t *cpuInfo)
```

Comment:
We could just extend the original API. The only thing that changes is the constraints on the inline asm.

I added a new function because OMR defines 2 different APIs for cpuid depending on the need for ECX, and there are 2 different intrinsics as well.